src/types/instantsearch.ts

```diff
@@ -10,6 +12,50 @@ export type SearchParameters = any;
 
 export type SearchResults = any;
 
+export type Hit = {
+  [attribute: string]: any;
+  objectID: string;
+  _highlightResult?: {
+    [attribute: string]: {
```

Comment:
This definition is not 100% correct. It's possible to have a recursive object or even an array.

```json
{
  "categories": [
    { "value": "", "matchLevel": "none" },
    { "value": "", "matchLevel": "none" }
  ]
}
```

```json
{
  "hierarchical": {
    "lvl0": {
      "value": "",
      "matchLevel": "none"
    }
  }
}
```

Reply:
would make sense to add HighlightResult as a type and then write all three possibilities with the result as an end factor

Reply:
Thanks! I fixed that in d551209ae664e717685ef7d43150d83c716f679c for both highlight and snippet.